examples/transformers/python_llama2/txtgen.py

```diff
@@ -0,0 +1,182 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#####################################################################################
+
+from argparse import ArgumentParser
+from transformers import LlamaTokenizer
+import numpy as np
+import migraphx as mgx
+import os
+import sys
+import time
+from functools import wraps
+
+
+# measurement helper
+def measure(fn):
+    @wraps(fn)
+    def measure_ms(*args, **kwargs):
+        start_time = time.perf_counter_ns()
+        result = fn(*args, **kwargs)
+        end_time = time.perf_counter_ns()
+        print(
+            f"Elapsed time for {fn.__name__}: {(end_time - start_time) * 1e-6:.4f} ms\n"
+        )
+        return result
+
+    return measure_ms
+
+
+def get_args():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "-p",
+        "--prompt",
+        type=str,
+        required=True,
+        help="Input prompt",
+    )
+
+    parser.add_argument(
+        "-l",
+        "--log-process",
+        action="store_true",
+        help="Print the current state of transcribing.",
+    )
+
+    parser.add_argument("-s",
+                        "--max-seq-len",
+                        type=int,
+                        choices=[256, 512, 1024, 2048, 4096],
+                        default=1024,
+                        help="Max sequence length the model can handle")
+
+    return parser.parse_args()
+
+
+class Llama2MGX():
+    def __init__(self, max_seq_len=1024):
+        model_id = "meta-llama/Llama-2-7b-chat-hf"
+        self.max_seq_len = max_seq_len
+
+        print("Load mgx model")
+        self.model = Llama2MGX.load_mgx_model(
+            max_seq_len, {
+                "input_ids": [1, max_seq_len],
+                "attention_mask": [1, max_seq_len],
+                "position_ids": [1, max_seq_len]
+            })
+        print(f"Load AutoTokenizer model from {model_id}")
+        self.tokenizer = LlamaTokenizer.from_pretrained(model_id)
+
+    @staticmethod
+    @measure
+    def load_mgx_model(max_seq_len, shapes):
+        file = "models/llama-2-7b-chat-hf/model"
```

Comment:
can you double check this is the name of the model when exporting with optimum? For me, I see that it's called "decoder_model"

Reply:
I've checked it and for me the name was `model.onnx`

Reply:
Looks like it may depend on what's being passed into optimum-cli. But with the arguments provided, it does seem to produce "model.onnx"